fe/fe-core/src/main/java/com/starrocks/connector/kudu/KuduMetadata.java

```diff
@@ -315,13 +321,24 @@ public Statistics getTableStatistics(OptimizerContext session,
 
         } else {
             try {
-                String kuduTableName = getKuduFullTableName(kuduTable.getDbName(), kuduTable.getTableName());
+                String kuduTableName = getKuduFullTableName(kuduTable);
                 rowCount = kuduClient.openTable(kuduTableName).getTableStatistics().getLiveRowCount();
+            } catch (RpcRemoteException e) {
+                if (isGetTableStatisticsUnsupported(e)) {
+                    LOG.warn("GetTableStatistics method not supported. Fallback to return default row count 1.");
+                    rowCount = 1;
+                } else {
+                    throw new RuntimeException("RPC error while getting table statistics", e);
+                }
             } catch (KuduException e) {
-                throw new RuntimeException(e);
+                throw new RuntimeException("Error while accessing Kudu table", e);
             }
         }
         builder.setOutputRowCount(rowCount);
         return builder.build();
     }
+
+    private static boolean isGetTableStatisticsUnsupported(RpcRemoteException e) {
+        return e.getMessage().contains("invalid method name: GetTableStatistics");
```

Comment:
it's a little trick. is there a better way to check this?

Reply:
I'm afraid no, kuduClient.isStatisticsEnabled() refers to a client-side configuration.